### PR TITLE
DOC: Improve to_parquet documentation

### LINKED ISSUE #29476: Improve fname documentation for to_parquet
#### Code Sample, a copy-pastable example if possible

Should to_parquet without file behave the same way than to_json in which if no file is specified we return the buffer?

### PR DESCRIPTION
**Pull request by larroy**
fixes #29476

- [x] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry  / NA

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Looks like this doc string actually lives in `pandas/core/frame.py` so its rendered here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_parquet.html

Otherwise looks good to me

File: pandas/io/parquet.py [hunk 1]
```diff
@@ -245,6 +245,17 @@ def to_parquet(
...
     kwargs
         Additional keyword arguments passed to the engine
+
+    Examples
+    --------
+    If you want get a buffer to the parquet content you can use a io.BytesIO object
+    >>> f = io.BytesIO()
+    >>> df.to_parquet(f)
+    >>> f.seek(0)
+    >>> len(f.read())
```
Comment (alimcmaster1) [inline pandas/io/parquet.py:255]: maybe just assign `f.read()` to a variable for ease of use. What do you think?

Comment (larroy) [reply pandas/io/parquet.py:255]: sure

File: pandas/core/frame.py [hunk 2]
```diff
@@ -2215,6 +2215,14 @@ def to_parquet(
            col1  col2
         0     1     3
         1     2     4
+
+        If you want get a buffer to the parquet content you can use a io.BytesIO object,
+        as long as you don't use partition_cols, which creates multiple files.
+
+        >>> f = io.BytesIO()
```
Comment (dsaxton) [inline pandas/core/frame.py:2222]: @larroy Looks like you need to import io first

File: pandas/core/frame.py [hunk 3]
```diff
@@ -2215,6 +2215,16 @@ def to_parquet(
...
+
+        If you want get a buffer to the parquet content you can use a io.BytesIO object,
+        as long as you don't use partition_cols, which creates multiple files.
+
+        >>> import io
+        >>> f = io.BytesIO()
+        >>> df.to_parquet(f)
+        >>> f.seek(0)
+        0
+        >>> content=f.read()
```
Comment (alimcmaster1) [inline pandas/core/frame.py:2227]: Nitpick but can you add whitespace around the = operator? I thought this would get picked up in our code checks..

**Comment by alimcmaster1** [comment]
LGTM - minor comment

File: pandas/core/frame.py (hunk 3 above)
Comment (larroy) [reply pandas/core/frame.py:2227]: yes, sure.

**Review by alimcmaster1 — APPROVED** [review]
LGTM cc @jreback

**Comment by alimcmaster1** [comment]
@pandas-dev/pandas-core to review/merge

File: pandas/core/frame.py [hunk 4]
```diff
@@ -2225,6 +2225,16 @@ def to_parquet(
            col1  col2
         0     1     3
         1     2     4
+
+        If you want get a buffer to the parquet content you can use a io.BytesIO object,
```
Comment (WillAyd) [inline pandas/core/frame.py:2229]:
```suggestion
        If you want to get a buffer to the parquet content you can use a io.BytesIO object,
```

**Review by WillAyd — COMMENTED** [review]
minor typo but looks good otherwise. If you can accept suggested edit should be OK to merge

**Comment by alimcmaster1** [comment]
Cc @WillAyd this looks good to me.

**Comment by WillAyd** [comment]
Thanks @larroy

**Comment by larroy** [comment]
Thank you all for the review and help.

**Pull request merged**